sdk/python/lib/pulumi/resource.py

```diff
@@ -632,7 +634,8 @@ def __init__(self,
                 props = tres.props
                 opts = tres.opts
 
-        self._name = name
+                if tres.name and isinstance(tres.name, str):
+                    self._name = tres.name
```

Comment:
Not sure we want a type test that silently ignores here - I think we just take what the user provided and overwrite it - if they disobey the types, they will get an error that makes sense to them.

Reply:
meaning there is type checking elsewhere in the framework that will prevent non-strings from being used (and throw a commonly used exception), or instead we show throw here?